### DISCUSSION TITLE
Add conversions to a queue batch

### DISCUSSION THREAD

**Discussion by driftingly** [spatie/laravel-medialibrary#2874] [Q&A]
Does anyone know of a way to get the conversion job instead of running it?

We have a job that creates a batch of images to download, we would like to add the conversions to the same batch so we can notify the user when the images finished downloading AND the conversion has run. 

Something like this:

```php
$image = $this->product
    ->addMediaFromUrl($url)
    ->toMediaCollection()
    ->withoutConversion();
    
$this->batch()->add($image->getConversion());
```

https://laravel.com/docs/master/queues#job-batching

**Comment by freekmurze** [comment]
Right now you can't get to the job itself. I would accept a PR that would put the conversion jobs into a batch and a PR that would allow users to get that batch and/or job.

**Comment by driftingly** [comment]
We ended up changing the `queue_conversions_by_default` setting to `false` in the batch job which allowed it to be run in the same process.  I may come back to this when I have more time to allow the batch to be passed in.
